### DISCUSSION TITLE
Pinch differences between v9 and v10

### DISCUSSION THREAD

**Discussion by agusterodin** [pmndrs/use-gesture#559] [Q&A]
Hey! I am converting somebody else's useGesture code from v9 to v10. I am attempting to get the usePinch portion working again.

I noticed in the [breaking changes list](https://github.com/pmndrs/use-gesture/blob/main/packages/react/CHANGELOG.md#1000) that "`state.movement` and `state.offset` now reflect scale and not distance". 

How do i emulate the old behavior of going off of distance? The value of scale appears to grow a lot differently (seems to grow exponentially instead of linearly now).

**Comment by dbismut** [comment]
Hi @agusterodin, what exactly are you trying to do? Are you still trying to zoom in and out? If so just use the scale values and forget about trying to map distance with a scale factor. New scale values are much more reliable. The demo folder is filled with examples that you can have a look at.

If you still want to use raw distance, then you can use `state.values[0]` but the gesture won't compute movement and offset for you and you'd have to do this manually.

**Reply by agusterodin** [reply]
Thanks for quick response 🙂

The use-case was origin corrected pinch to zoom. I ended up ripping the old code out and using the implementation from this example: https://codesandbox.io/embed/github/pmndrs/use-gesture/tree/main/demo/src/sandboxes/card-zoom?codemirror=1
